config/esp32/toolchain/BUILD.gn

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+
+import("//build/toolchain/gcc_toolchain.gni")
+
+declare_args() {
+  ar_ext = ""
+  cc_ext = ""
+  cxx_ext = ""
+  cpu_ext = ""
+}
+
+gcc_toolchain("esp32") {
+  ar = ar_ext
+  cc = cc_ext
+  cxx = cxx_ext
+
+  toolchain_args = {
+    current_os = "freertos"
+    current_cpu = cpu_ext
```

Comment:
Can this be "xtensa" ? The cpu variable is typically used to determine the architecture.

Reply:
I've been going back and forth on that myself. Finally I thought there is nothing 'xtensa' specific here, the toolchain is esp32-specific, the SDK is esp32-specific, so might as well call it 'esp32'. Thoughts? What typically goes into generic 'cpu' specific files?

Reply:
The use case is writing assembly code. See e.g. 
https://github.com/project-chip/connectedhomeip/blob/master/third_party/jlink/segger_rtt/RTT/BUILD.gn#L40